tests/providers/amazon/aws/operators/test_eks.py

```diff
@@ -0,0 +1,299 @@
+    @mock.patch.object(EKSHook, "delete_cluster")
+    def test_existing_cluster_not_in_use(self, mock_delete_cluster, mock_list_nodegroups):
+        mock_list_nodegroups.return_value = dict(nodegroups=list())
+
+        self.delete_cluster_operator.execute({})
+
+        mock_list_nodegroups.assert_called_once
+        mock_delete_cluster.assert_called_once_with(name=self.cluster_name)
+
+
+class TestEKSDeleteNodegroupOperator(unittest.TestCase):
+    def setUp(self) -> None:
+        self.cluster_name: str = random_names()
+        self.nodegroup_name: str = random_names()
+
+        self.delete_nodegroup_operator = EKSDeleteNodegroupOperator(
+            task_id=TASK_ID, cluster_name=self.cluster_name, nodegroup_name=self.nodegroup_name
+        )
+
+    @mock.patch.object(EKSHook, "delete_nodegroup")
+    def test_existing_nodegroup(self, mock_delete_nodegroup):
+        self.delete_nodegroup_operator.execute({})
+
+        mock_delete_nodegroup.assert_called_once_with(
+            clusterName=self.cluster_name, nodegroupName=self.nodegroup_name
+        )
+
+
+class TestEKSDescribeAllClustersOperator(unittest.TestCase):
+    def setUp(self) -> None:
+        self.describe_all_clusters_operator = EKSDescribeAllClustersOperator(task_id=TASK_ID)
+
+    @mock.patch.object(EKSHook, "list_clusters")
+    @mock.patch.object(EKSHook, "describe_cluster")
+    def test_clusters_exist_returns_all_cluster_details(self, mock_describe_cluster, mock_list_clusters):
+        cluster_names: List[str] = NAME_LIST
+        response = dict(clusters=cluster_names, nextToken=DEFAULT_NEXT_TOKEN)
+        mock_describe_cluster.return_value = EMPTY_CLUSTER
+        mock_list_clusters.return_value = response
+
+        self.describe_all_clusters_operator.execute({})
+
+        mock_list_clusters.assert_called_once()
+        assert mock_describe_cluster.call_count == len(cluster_names)
+
+    @mock.patch.object(EKSHook, "list_clusters")
+    @mock.patch.object(EKSHook, "describe_cluster")
+    def test_no_clusters_exist(self, mock_describe_cluster, mock_list_clusters):
+        mock_list_clusters.return_value = dict(clusters=list(), token=DEFAULT_NEXT_TOKEN)
+
+        self.describe_all_clusters_operator.execute({})
+
+        mock_list_clusters.assert_called_once()
+        mock_describe_cluster.assert_not_called()
+
+
+class TestEKSDescribeAllNodegroupsOperator(unittest.TestCase):
+    def setUp(self) -> None:
+        self.cluster_name: str = random_names()
+
+        self.describe_all_nodegroups_operator = EKSDescribeAllNodegroupsOperator(
+            task_id=TASK_ID, cluster_name=self.cluster_name
+        )
+
+    @mock.patch.object(EKSHook, "list_nodegroups")
+    @mock.patch.object(EKSHook, "describe_nodegroup")
+    def test_nodegroups_exist_returns_all_nodegroup_details(
+        self, mock_describe_nodegroup, mock_list_nodegroups
+    ):
+        nodegroup_names: List[str] = NAME_LIST
+        cluster_name: str = random_names()
```

Comment:
Is it needed? Why can't you use a constant value?

Reply:
Removed in upcoming revision.

Reply:
Corrected in https://github.com/apache/airflow/pull/16571/commits/76c7ad635948191764a323a9be03fa965fc91ff5